core/input_map.h

```diff
@@ -39,25 +39,50 @@ class InputMap : public Object {
 	GDCLASS(InputMap, Object);
 
 public:
+	enum ActionPlayer {
+		PLAYER_ALL = 0,
+		PLAYER_1, // 1
+		PLAYER_2,
+		PLAYER_3,
+		PLAYER_4,
+		PLAYER_5,
+		PLAYER_6,
+		PLAYER_7,
+		PLAYER_8,
+		PLAYER_9,
+		PLAYER_10,
+		PLAYER_11,
+		PLAYER_12,
+		PLAYER_13,
+		PLAYER_14,
+		PLAYER_15,
+		PLAYER_16
+	};
+
 	/**
 	* A special value used to signify that a given Action can be triggered by any device
 	*/
 	static int ALL_DEVICES;
 
+	struct ActionInput {
+		ActionPlayer player;
```

Comment:
Just out of curiousity, will this limit the number of players to 16, whatever counts as a player to the engine? Or will it still allow arbitrary numbers to be input here?

Reply:
It somehow limits the number of split-screen players, or local players with same input actions but different input events for such actions. It does not in any way the limit the number of players in a broader sense.

> And since I'm here, I should also ask, what counts as a "player" to the engine?

It's just a number used to filter input events.
So you can have the same action names with different input events.
```gdscript
# Check if ui_left is pressed with the key associated to player 1 in InputMap
InputMap.is_action_pressed("ui_left", 1)

# Check if ui_left is pressed with the key associated to player 2 in InputMap
InputMap.is_action_pressed("ui_left", 2)

# Check if ui_left is pressed with any key configured in InputMap
InputMap.is_action_pressed("ui_left", 0)
```

Reply:
>It somehow limits the number of split-screen players, or local players with same input actions but different input events for such actions. It does not in any way the limit the number of players in a broader sense.

Perhaps I worded my question wrong. I was asking if that number could be a number higher than 16, without having to edit and recompile the engine. Although I'll admit that the needs for a number higher than 16 are few and far between (if one even exists), so chances are that this question is moot.

>It's just a number used to filter input events.

But what does that number refer to? A controller index?

Reply:
> Perhaps I worded my question wrong. I was asking if that number could be a number higher than 16

Yes, it does limit it to 16, but as you said, it's very unlikely you need more that that.

> What does that number refer to? A controller index?

No, the local order you would give your players. And the value you set in `InputMap` or pass to `action_add_event`